package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -1146,7 +1146,7 @@ dnsmasq_start()
 
 
 	# Create a dnsmasq.d dir for each instance
-	config_get dnsmasqconfdir "$cfg" confdir "/tmp/dnsmasq${cfg:+.$cfg}.d"
+	config_get dnsmasqconfdir "$cfg" confdir "/tmp/dnsmasq.d"
```

Comment:
Change to
```sh
	# Create a dnsmasq.d dir for each instance
	if [ "$cfg" = "$DEFAULT_INSTANCE" ]; then
		config_get dnsmasqconfdir "$cfg" confdir "/tmp/dnsmasq.d"
	else
		config_get dnsmasqconfdir "$cfg" confdir "/tmp/dnsmasq${cfg:+.$cfg}.d"
	fi

Reply:
How would other packages figure out what's the default confdir per instance then?

Reply:
Either `confdir` set per instance, or the instance name (the 'name' part that follows the `config dnsmasq` in dhcp).

Bear in mind that it is still possible for a user to set `confdir` on the main instance which would (still) break packages expecting `/tmp/dnsmasq.d` as the default. And that's no different to how it's been for a long time.


This is why I think a `/tmp/dnsmasq.common.conf` would be useful. Something shared between all instances.

Reply:
I'm talking about when confdir is NOT set per instance. If the default values are different between the DEFAULT_INSTANCE and other instances, how would the packages know which is which?

Reply:
/tmp/dnsmasq.d is the default for DEFAULT_INSTANCE. If it isn't that folder, it's another instance.

Reply:
Or are you asking how to reconcile an instance to its folder?

Reply:
If so, then a common conf is doable, even in ujail'd dnsmasq.

```sh
	# Define a common dnsmasq configuration file
	DNSMASQ_COMMON_FILE="/etc/dnsmasq.common.conf"

	# Create a dnsmasq.d dir for each instance
	if [ "$cfg" = "$DEFAULT_INSTANCE" ]; then
		config_get dnsmasqconfdir "$cfg" confdir "/tmp/dnsmasq.d"
	else
		config_get dnsmasqconfdir "$cfg" confdir "/tmp/dnsmasq${cfg:+.$cfg}.d"
	fi
	# Ensure dnsmasqconfdir is an absolute path
	[ "${dnsmasqconfdir:0:1}" = '/' ] && {
		# Include the common config file if it exists
		if [ -d "$DNSMASQ_COMMON_FILE" ]; then
			xappend "--conf-file=$DNSMASQ_COMMON_FILE"
		fi
		# Include the instance-specific config dir
		xappend "--conf-dir=$dnsmasqconfdir"
		dnsmasqconfdir="${dnsmasqconfdir%%,*}"
		[ ! -d "$dnsmasqconfdir" ] && mkdir -p "$dnsmasqconfdir"
		xappend "--user=dnsmasq"
		xappend "--group=dnsmasq"
		echo >> "$CONFIGFILE_TMP"

		# EXTRACONFFILE allows new dnsmasq parameters before they are natively handled in this init file
		config_get extraconftext "$cfg" extraconftext
		[ -n "$extraconftext" ] && echo -e "$extraconftext" > "$dnsmasqconfdir"/"$EXTRACONFFILE"
	}
```

The common file starts up empty, so just dump whatever is needed common across all instances there.